### PR TITLE
Add uptime monitoring every min and every 5 secs if minute ping fails

### PR DESCRIPTION
**Pull request by marianamarasoiu**
This also adds a snackbar/toast that could be triggered from the platform. I initially did this so I can trigger the snackbar to show failures, but since it's so intrusive, I ended up using the banner notification instead.

### CODE REVIEW

File: webapp/lib/pubsub.dart [hunk 1]
```diff
@@ -20,8 +20,8 @@ class PubSubClient extends DocPubSubUpdate {
 
   PubSubClient(this.publishUrl, this.user);
 
-  /// Publish the specified exception.
-  /// Callers should catch and handle IOException.
```
Comment (danrubel) [inline webapp/lib/pubsub.dart:24]: `PubSubException` is only one of the exceptions that can be thrown here.
Perhaps `Callers should catch and handle PubSubException and IOException` instead?

Comment (marianamarasoiu) [reply webapp/lib/pubsub.dart:24]: Good point, thanks. But I don't think IO exception exists for web code? IIRC, this comment was left out of date from when we accidentally imported `dart:io` and had `PubSubException` inherit from `IOException`

Looking at the `http` library, it looks like the only exception they define is `ClientException`, so I've added that instead. If there's another exception that would be good to list here, please let me know.

File: webapp/lib/platform.dart [hunk 2]
```diff
@@ -42,6 +43,129 @@ init() async {
...
+    });
+  };
+
+  new Timer.periodic(oneMinute, (Timer t) {
+    Map payload = {
+      'ping': '${t.tick}',
+      'lastUserActivity': controller.lastUserActivity.toIso8601String()
+    };
+    _uptimePubSubInstance.publish(platform_constants.statuszTopic, payload).then(
+      (_) {
```
[the remark this answers is not in the document]
Comment (marianamarasoiu) [reply webapp/lib/platform.dart:120]: Good idea, thanks, done!

File: webapp/lib/pubsub.dart (hunk 1 above)
Comment (danrubel) [reply webapp/lib/pubsub.dart:24]: @marianamarasoiu Ah yes. That sounds right. Thanks!

**Comment by marianamarasoiu** [comment]
Thanks both for the review! merging now..

**Pull request merged**